report.md

```diff
@@ -29,36 +29,47 @@ The eight functions are presented in the table below. Every function has a CCN o
-| cycle_sort(list)                                                    | sorts/cycle_sort.py              | 10          | 10                       | -                       | 35           |
-| spiral_print(matrix)                                                           | matrix/spiral_print.py                        | 12        | 10                | -                | NLOC5        |
+| interpolation_search(sorted_collection, item)                       | searches/interpolation_search.py | 10          | 6                       | -                       | 30           |
+| cycle_sort(list)                                                    | sorts/cycle_sort.py              | 10          | 10                      | -                       | 35           |
+| spiral_print(matrix)                                                | matrix/spiral_print.py           | 12          | 10                      | -                       | NLOC5        |
 | Function6                                                           | Location6                        | CCN6        | CCN6man1                | CCN6man2                | NLOC6        |
 | Function7                                                           | Location7                        | CCN7        | CCN7man1                | CCN7man2                | NLOC7        |
 | Function8                                                           | Location8                        | CCN8        | CCN8man1                | CCN8man2                | NLOC8        |
 
 - **Did all methods (tools vs. manual count) get the same result?**
 - **Are the results clear?**
 
-2. **Are the functions just complex, or also long?**
-3. **What is the purpose of the functions?**
-4. **Are exceptions taken into account in the given measurements?**
-5. **Is the documentation clear w.r.t. all the possible outcomes?**
+1. **Are the functions just complex, or also long?**
+2. **What is the purpose of the functions?**
+3. **Are exceptions taken into account in the given measurements?**
+4. **Is the documentation clear w.r.t. all the possible outcomes?**
+
+### <a id="dj_oracle"></a>Deutsch Jozsa ([file](quantum/deutsch_jozsa.py))
+
+The Deutsch-Jozsa problem features an oracle function that takes an n-bit input and outputs 0 or 1. The function is either constant (returns a constant value) or balanced (meaning it returns 0 half of the time and 1 the rest of the time). The purpose of the examined method is to create a Quantum Cirtcuit (a model) of an oracle function which is either balanced or constant depending on the parameter `case`, and that handles `num_qubits`-bit input values.
+
+The function handles multiple cases and needs to iterate through `num_qubits` when setting the output values for the model. At the same time, the only way to exit the function is to return the model. This is why the cyclomatic complexity is so high. The documentation is clear about the only possible outcome of the function.
```

Comment:
Can you explain more precisely what are the possible outputs of the algorithm ?

Reply:
Sure thing